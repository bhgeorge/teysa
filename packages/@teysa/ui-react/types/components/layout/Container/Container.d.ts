import { ReactNode } from 'react';
declare type ContainerSize = 'sm' | 'md' | 'lg';
interface ContainerProps {
    size?: ContainerSize;
    children: ReactNode | ReactNode[];
}
declare const Container: ({ size, children }: ContainerProps) => JSX.Element;
export default Container;
