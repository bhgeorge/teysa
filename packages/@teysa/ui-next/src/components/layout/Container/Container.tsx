// Global
import React, { ReactNode } from 'react';
import classNames from 'classnames';

type ContainerSize = 'sm' | 'md' | 'lg';

interface ContainerProps {
  size?: ContainerSize;
  children: ReactNode | ReactNode[];
}

const sizeClasses: Record<ContainerSize, string> = {
  sm: 'max-w-prose',
  md: 'max-w-screen-md',
  lg: 'max-w-screen-lg',
};

export function Container({ size = 'md', children }: ContainerProps) {
  return (
    <div
      className={classNames(
        'w-full',
        'mx-auto',
        'px-gutter',
        sizeClasses[size]
      )}
    >
      {children}
    </div>
  );
}
