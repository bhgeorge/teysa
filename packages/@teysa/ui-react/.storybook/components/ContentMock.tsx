// Global
import React from 'react';

const ContentMock = () => (
  <div className="p-8 flex items-center justify-center w-full bg-theme-text text-theme-bg">
    content
  </div>
);

export default ContentMock;
