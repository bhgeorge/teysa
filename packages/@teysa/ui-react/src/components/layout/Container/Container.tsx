// Global
import React, { ReactNode } from 'react';
import classNames from 'classnames';

type ContainerSize = 'sm' | 'md' | 'lg';

interface ContainerProps {
  size?: ContainerSize;
  children: ReactNode | ReactNode[];
}

const sizeClasses: Record<ContainerSize, string> = {
  sm: 'max-w-prose',
  md: 'max-w-screen-md',
  lg: 'max-w-screen-lg',
};

const Container = ({ size = 'md', children }: ContainerProps) => (
  <div className={classNames('w-full', 'mx-auto', 'px-gutter', sizeClasses[size])}>{children}</div>
);

export default Container;
