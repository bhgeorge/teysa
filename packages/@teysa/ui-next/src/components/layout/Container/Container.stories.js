// Storybook
import ContentMock from '../../../../.storybook/components/ContentMock';
// Local
import { Container } from './Container';

export default {
  title: 'Layout/Container',
  component: Container,
};

const Template = props => (
  <Container {...props}>
    <ContentMock />
  </Container>
);

export const Default = Template.bind({});
