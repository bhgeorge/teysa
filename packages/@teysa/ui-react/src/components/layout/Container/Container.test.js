// Global
import { render } from '@testing-library/react';
import React from 'react';
// Storybook
import ContentMock from '../../../../.storybook/components/ContentMock';
// Local
import Container from './Container';

const snapshot = (size) => {
  const component = render(
    <Container size={size}>
      <ContentMock />
    </Container>
  );

  expect(component.container.firstChild).toMatchSnapshot();
};

it('renders correctly at all sizes', () => {
  snapshot('sm');
  snapshot('md');
  snapshot('lg');
});
