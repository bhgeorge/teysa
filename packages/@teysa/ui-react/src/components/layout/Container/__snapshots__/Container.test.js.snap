// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly at all sizes 1`] = `
<div
  class="w-full mx-auto px-gutter max-w-prose"
>
  <div
    class="p-8 flex items-center justify-center w-full bg-theme-text text-theme-bg"
  >
    content
  </div>
</div>
`;

exports[`renders correctly at all sizes 2`] = `
<div
  class="w-full mx-auto px-gutter max-w-screen-md"
>
  <div
    class="p-8 flex items-center justify-center w-full bg-theme-text text-theme-bg"
  >
    content
  </div>
</div>
`;

exports[`renders correctly at all sizes 3`] = `
<div
  class="w-full mx-auto px-gutter max-w-screen-lg"
>
  <div
    class="p-8 flex items-center justify-center w-full bg-theme-text text-theme-bg"
  >
    content
  </div>
</div>
`;
